presto-local-file/src/main/java/com/facebook/presto/localfile/LocalFileRecordCursor.java

```diff
@@ -0,0 +1,323 @@
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.zip.GZIPInputStream;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.TimestampType.TIMESTAMP;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toSet;
+
+public class LocalFileRecordCursor
+        implements RecordCursor
+{
+    private static final Splitter LINE_SPLITTER = Splitter.on("\t").trimResults();
+
+    // TODO This should be a config option as it may be different for different log files
+    public static final DateTimeFormatter ISO_FORMATTER = new DateTimeFormatterBuilder()
+            .append(ISODateTimeFormat.dateHourMinuteSecondFraction())
+            .appendTimeZoneOffset("Z", true, 2, 2)
+            .toFormatter();
+
+    private final int[] fieldToColumnIndex;
+    private final HostAddress address;
+    private final List<LocalFileColumnHandle> columns;
+    private final FilesReader reader;
+    private final boolean includeServer;
+
+    private List<String> fields;
+
+    public LocalFileRecordCursor(LocalFileTables localFileTables, List<LocalFileColumnHandle> columns, SchemaTableName tableName, HostAddress address, TupleDomain<LocalFileColumnHandle> predicate)
+    {
+        this.columns = requireNonNull(columns, "columns is null");
+        this.address = requireNonNull(address, "address is null");
+
+        fieldToColumnIndex = new int[columns.size()];
+        for (int i = 0; i < columns.size(); i++) {
+            LocalFileColumnHandle columnHandle = columns.get(i);
+            fieldToColumnIndex[i] = columnHandle.getOrdinalPosition();
+        }
+        this.includeServer = isServerIncluded(address, predicate, localFileTables.getTable(tableName));
+        this.reader = includeServer ? getFilesReader(localFileTables, predicate, tableName) : null;
+    }
+
+    private static boolean isServerIncluded(HostAddress address, TupleDomain<LocalFileColumnHandle> predicate, LocalFileTableHandle table)
+    {
+        if (!table.getServerAddressColumn().isPresent()) {
+            return true;
+        }
+
+        Optional<Map<LocalFileColumnHandle, Domain>> domains = predicate.getDomains();
+        if (!domains.isPresent()) {
+            return true;
+        }
+
+        Set<Domain> serverAddressDomain = domains.get().entrySet().stream()
+                .filter(entry -> entry.getKey().getOrdinalPosition() == table.getServerAddressColumn().getAsInt())
```

Comment:
using the `ordinalPosition == -1` seems a bit brittle.  Maybe we should add an enum to the  `LocalFileColumnHandle` that says what kind of column we have